### PR TITLE
Dev/179 Actions in the New Problem Editor

### LINKED ISSUE #179: New Problem Editor: Archive Problem.

### PR DESCRIPTION
**Pull request by Chaoyuuu**
Closed #179 

image

image

### CODE REVIEW

File: src/app/problem/editor/right/Actions.js [hunk 1]
```diff
@@ -1,8 +1,102 @@
...
+    const {markProblemsDirty} = useProblemEditorContext();
+    const [archived, setArchived] = useState(false);
+    const [showDeleteProblemModal, setShowDeleteProblemModal] = useState(false);
+
+    useEffect(() => {
+        if (!problem) {
+            problemService.getProblemById(problemId)
+                .then(p => {
+                    setProblem(p);
+                    setArchived(p.archived)
```
Comment (Johnny850807) [inline src/app/problem/editor/right/Actions.js:31]: semicolon

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**